### PR TITLE
[Dotenv] add json output to DebugCommand.php

### PR DESCRIPTION
**Pull request by jonashrem**
adds an optional json output to 'debug:dotenv' command

| Q             | A
| ------------- | ---
| Branch?       | 6.1 
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| License       | MIT
| Doc PR        |  /

### CODE REVIEW

**Comment by ro0NL** [comment]
curious, why do you need a JSON machine format .. compared to current PHP machine format (.env.local.php)

perhaps the `dotenv:dump` command should be tweaked instead

**Comment by jonashrem** [comment]
> curious, why do you need a JSON machine format .. compared to current PHP machine format (.env.local.php)
> 
> perhaps the `dotenv:dump` command should be tweaked instead

To read information from external projects without changing them.

For my understanding, dotenv:dump writes files to a .env.local.php inside the project directory which is not always desired.

In my case, I want to get information from a project configuration without changing the project as I'm not in charge for the project.

A machine-readable output in  DebugCommand allows me to get information from any symfony project without needing to check if there is an .env file or .env.local.php (or something else) from outside.

**Comment by GromNaN** [comment]
Other `debug:*` commands use the [`DescriptorInterface`](https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/Console/Descriptor/DescriptorInterface.php). Adding a new output format is a great opportunity to refactor and use this feature.

File: src/Symfony/Component/Dotenv/Command/DebugCommand.php [hunk 1]
```diff
@@ -47,10 +48,21 @@ public function __construct(string $kernelEnvironment, string $projectDirectory)
         parent::__construct();
     }
 
+    protected function configure()
+    {
+        $this
+            ->setDescription(self::$defaultDescription)
```
Comment (derrabus) [inline src/Symfony/Component/Dotenv/Command/DebugCommand.php]: Setting the description here should not be necessary.

```suggestion
```

File: src/Symfony/Component/Dotenv/Command/DebugCommand.php [hunk 2]
```diff
@@ -63,6 +75,14 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             return is_file($this->getFilePath($file));
         });
 
+        if ('json' == $input->getOption('format')) {
+            $io->writeln(json_encode($this->getVariablesasArray($availableFiles), \JSON_PRETTY_PRINT | \JSON_UNESCAPED_SLASHES));
```
Comment (derrabus) [inline src/Symfony/Component/Dotenv/Command/DebugCommand.php]:
```suggestion
            $io->writeln(json_encode($this->getVariablesasArray($availableFiles), \JSON_PRETTY_PRINT | \JSON_UNESCAPED_SLASHES | \JSON_THROW_ON_ERROR));
```

**Review by derrabus — COMMENTED** [review]
Leveraging the descriptors as suggested seems like a good idea to me.

**Comment by derrabus** [comment]
By the way, we need tests for this feature.

**Comment by fabpot** [comment]
Closing for now as it should leverage the already existing infrastructure, which would make a very different PR.

**Pull request closed without merging**